### PR TITLE
refactor gulp build code

### PR DESCRIPTION
**Pull request by giladgray**
decided to give the gulp code a little polishing ✨ 

- remove unused `getTypescriptSources` (replaced by globs)
- remove `blueprint.dest()`: unnecessary complexity
- remove `blueprint.taskMapper()`: use `projects.map()` to build array once and avoid duplicating strings
- move karma config to separate file (🎩 @codebling)
- delete unused (and incomplete) mocha task
- add a missing license block

### CODE REVIEW

File: gulp/index.js [hunk 1]
```diff
@@ -8,27 +8,10 @@ const plugins = require("gulp-load-plugins")();
 const assign = require("lodash/assign");
```
Comment (adidahiya) [inline gulp/index.js]: unused import?

**Comment by giladgray** [comment]
I have an even sweeter refactor ready to go on top of this that hopefully resolves a lot of the ambient pain with this gulp build by making things much more explicit. But I'd like to open that as a separate review, so let's please get this one reviewed ASAP cuz it's still got some good relevant cleanup (like removing all those obsolete `blueprint.*` functions).

File: gulp/karma.js [hunk 2]
```diff
@@ -111,5 +33,7 @@ module.exports = (gulp, plugins, blueprint) => {
         });
     });
 
-    gulp.task("karma", (done) => rs(...blueprint.taskMapper("karma", "karma-"), done));
+    // running in sequence so output is human-friendly
+    // (in parallel, all suites get interspersed and it's a mess)
```
Comment (adidahiya) [inline gulp/karma.js]: for another PR: try running expensive parallel tasks in separate CircleCI containers

File: gulp/util/task.js [hunk 3]
```diff
@@ -1,3 +1,6 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
```
Comment (adidahiya) [inline gulp/util/task.js]: 2016?

File: gulp/util/karma-config.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
```
Comment (adidahiya) [inline gulp/util/karma-config.js]: 2016?

**Comment by giladgray** [comment]
pushed some last-minute tweaks that came up in my other feature work but make sense to get in here. review the last few commits individually for best results.

**Pull request merged**